### PR TITLE
Ex1

### PR DESCRIPTION
**Pull request by lahavrom**
DoDo App- The app that will help you do.
In this app you can add, track and delete tasks you need to do.
Features:
- adding tasks by pressing the Enter key
- input validations (empty task, same task)
- sorting your tasks by their names
- sorting your tasks by their creation date
- each sort button is both for ascending and descending orders- just press it again
- "Clear All" button for deleting all of your tasks
- cool animations

@eranbenmoha @amirbardugo

### CODE REVIEW

File: src/ex1/script.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        return;
+    }
+    if (document.getElementById(task.value) !== null){
+        alert("You already have that task!");
+        task.value = '';
+        return;
+    }
+    if (task.value){
+        // create div for task + delete button
+        const task_section = document.createElement("div");
```
Comment (eranbenmoha) [inline src/ex1/script.js:40]: I suggest you move the task element creation to a separate function, for readability and reusability

File: src/ex1/script.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        deleteIcon.className = "fa fa-trash-o";
+        delete_btn.appendChild(deleteIcon);
+        document.getElementById(task.value).appendChild(delete_btn);
+        // delete input
+        task.value = '';        
+    }
+};
+
+// returns the element that shows the date & time the task created
+function taskTime(){
```
Comment (eranbenmoha) [inline src/ex1/script.js:55]: nice touch!

File: src/ex1/script.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
+
+function addTask(){
+    const task = document.getElementById('newTask');
+    // check if entered empty task
+    if (task.value === ''){
+        alert("You forgot to type the task!");
+        return;
+    }
+    if (document.getElementById(task.value) !== null){
```
Comment (eranbenmoha) [inline src/ex1/script.js]: Having the id of the DOM element as the id of the task AND the value presents a limitation
1. A user should be able to enter the same task multiple times
2. I can't enter tasks with the same value as existing DOM elements
 
A better solution would be to save the tasks in a list in memory with unique ids 

image

File: src/ex1/style.css [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    opacity: 0.7;
+}
+
+input {
+    padding: 12px;
+    width: 80%;
+    font-family: cursive;
+}
+.addBtn {
+    border-radius: 4px;
```
Comment (eranbenmoha) [inline src/ex1/style.css:33]: most of the buttons share the same CSS properties- it's better to have a standard button class (same color, bg color, border radius etc) and add another class for specific use cases

File: src/ex1/script.js (hunk 3 above)
Comment (lahavrom) [reply src/ex1/script.js]: yeah I did that on purpose... doesn't make sense- fixed it

**Pull request merged**